internal/services/v1/relationships.go

```diff
@@ -126,30 +130,49 @@ func (ps *permissionServer) ReadRelationships(req *v1.ReadRelationshipsRequest,
 		return rewriteError(ctx, err)
 	}
 
-	usagemetrics.SetInContext(ctx, &dispatchv1.ResponseMeta{
-		DispatchCount: 1,
-	})
+	batchCount := datastoreQueryBatchSize
+	var cursor options.Cursor
+	var dispatchCount uint32
+	for batchCount == datastoreQueryBatchSize {
```

Comment:
this can be tackled in a different PR since this is a more widespread problem, but for loops like this that incur expensive I/O operations, we want to check context cancelation. It would help the process free resources faster and spare wasteful I/O after cancelation is signaled.